drivers/stepper/adi_tmc/adi_tmc5041_stepper_controller.c

```diff
@@ -103,20 +106,30 @@ static void calculate_velocity_from_hz_to_fclk(const struct device *dev, const u
 		velocity_hz, *velocity_fclk);
 }
 
-static void set_async_signal(const struct device *dev, struct k_poll_signal *async)
+#ifdef CONFIG_STEPPER_ASYNC_API
+
+static int tmc5041_stepper_set_callback(const struct device *dev, stepper_callback_t  callback,
+					struct k_poll_signal *async)
```

Comment:
@bjarki-andreasen does this seem fine to you? The end user could attach a callback(probably distribute the events in the system using a zbus channel) and also "wait(if required)" with this particular function.

Reply:
to be honest this looks a bit off, the only thing I'd expect a "set_callback" function to take is a callback, would it make sense to have separate apis? (I'm admittedly not very familiar with who the poll api is meant to be used, wondering if it would not have been better to only implement a callback and let any poll signal logic been implemented by the application)

Reply:
Yeah that actually make a whole lot of sense to pass just an event callback and then the application can take care of POLLING, ZBUSSING that information or prolly something else. I will drop the support for k_poll_signal and if required could be lateron added as a dedicated function.